### ISSUE TITLE
False positives with coalescing assignment (PhpUnusedLocalVariableInspection and OnlyWritesOnParameterInspection)

### ISSUE THREAD

**Issue by uncaught** [kalessil/phpinspectionsea#2008]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 6.2.0 |
| Language level | PHP 8.5.5                       |

Current behaviour
--
This code triggers `PhpUnusedLocalVariableInspection` and `OnlyWritesOnParameterInspection`:

```php
      $isWorkDayCache = [];
      $isWorkDay = static fn(string $isoDate) => $isWorkDayCache[$isoDate] ??= $isWpWorkDay($isoDate);
```

Screenshot:

<img width="777" height="48" alt="Image" src="https://github.com/user-attachments/assets/00c8b8aa-61aa-470e-9f9b-82b9e92fa515" />

Expected behaviour
--

The code is valid and the variable is in use here. The rules should not be triggered.

**Comment by ea-inspections-team** [comment]
Thank you for reporting @uncaught!
The false positive has been resolved.
